eos_data_distribution/ndn/command.py

```diff
@@ -0,0 +1,90 @@
+# -*- Mode:python; coding: utf-8; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+#
+# Copyright (C) 2016 Endless Mobile INC.
+# Author: Niv Sardi <xaiki@endlessm.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# A copy of the GNU Lesser General Public License is in the file COPYING.
+
+import logging
+
+from pyndn.control_parameters import ControlParameters
+from pyndn.encoding.tlv_wire_format import TlvWireFormat
+from pyndn.util.command_interest_generator import CommandInterestGenerator
+from pyndn import Interest, Name
+
+logger = logging.getLogger(__name__)
+
+_commandInterestGenerator = CommandInterestGenerator()
+
+def generateInterest(*args, **kwargs):
+    logger.debug("args kwargs: %s %s", args, kwargs)
+    _commandInterestGenerator.generate(*args, **kwargs)
+
+def addNextHop(faceUri, name, cost=0, *args, **kwargs):
+    face = Face(faceUri)
+
+    controlParameters = ControlParameters()
+    if cost: controlParameters.setCost(int(cost))
+    interest = makeInterest('/nfd/fib/add-nexthop', name,
+                            controlParameters=controlParameters,
+                            *args, **kwargs)
+    face.expressInterest(interest)
+    return face
+
+def makeInterest(cmd, local=True, controlParameters={},
+                 keyChain=None, certificateName=None):
+    assert(cmd.startswith('/'))
+
+    commandInterest = Interest()
+    if local:
+        commandInterest.setName(Name("/localhost%s" % cmd))
+        # The interest is answered by the local host, so set a short timeout.
+        commandInterest.setInterestLifetimeMilliseconds(2000.0)
+    else:
+        commandInterest.setName(Name("/localhop%s" % cmd))
+        # The host is remote, so set a longer timeout.
+        commandInterest.setInteresLifetimeMilliseconds(4000.0)
+    # NFD only accepts TlvWireFormat packets.
+    commandInterest.getName().append(controlParameters.wireEncode(TlvWireFormat.get()))
+    generateInterest(commandInterest,
+                     keyChain, certificateName,
+                     TlvWireFormat.get())
+
+    return commandInterest
+
+if __name__ == '__main__':
+    from gi.repository import GLib
+    from . import base
+    import argparse
+
+    logging.basicConfig(level=logging.DEBUG)
+
+    parser = argparse.ArgumentParser(description='Command Interest Tests')
```

Comment:
Why not use `unittest` for the tests, then they can be run automatically when setuptools are run and when the package is built?